frontend/webservice/admin/admin.cgi

```diff
@@ -2559,16 +2567,52 @@ sub decom_workgroup{
     my ($method, $args) = @_;
 
     # my ($user, $err) = authorization(admin => 1, read_only => 0);
+    #Intial check to see if user is a normal system admin
     my ($result, $err) = OESS::DB::User::has_system_access(db => $db2, username => $ENV{'REMOTE_USER'}, role => 'normal'); 
     if (defined $err) {
         $method->set_error($err);
         return;
     }
-
+    #Find the workgroup to be decommissioned
     my $workgroup_id = $args->{'workgroup_id'}{'value'};
-    my $results;
+    my $workgroup = new OESS::Workgroup(db => $db2, workgroup_id => $workgroup_id);
+
+    if (!defined $workgroup) {
+        $method->set_error("No workgroup with that ID found");
+        return;
+    }
+    #If the workgroup to be decommissioned is an admin type workgroup escalate the permissions needed to high admin
+    if ($workgroup->{type} eq 'admin'){
+        ($result, $err) = OESS::DB::User::has_system_access(db => $db2, username => $ENV{'REMOTE_USER'}, role => 'admin');
```

Comment:
This is the second call to has_system_access. Instead why not get the workgroup type and use that as the argument to pass to role?